othello/moderator/moderator.py

```diff
@@ -78,13 +65,12 @@ def make_move(self, move):
                 )
                 board[self.current_player] |= c
                 board[opponent] &= utils.bit_not(c)
-        self.board, self.current_player = board, opponent
+        self.board = board
+        self.toggle_current_player()
 
     def check_game_over(self):
-        current_moves = self.possible_moves()
-        self.current_player ^= 1
-        opponent_moves = self.possible_moves()
-        self.current_player ^= 1
+        current_moves = self.possible_moves(player=self.current_player)
+        opponent_moves = self.possible_moves(player=1 ^ self.current_player)
         return self.board[0] & self.board[1] == constants.FULL_BOARD or not (
```

Comment:
Now that `possible_moves()` can be called without mutating data in-place, you can move the calls to that function down here and make this a single expression. That avoids calling `possible_moves()` if the board is full.